testapps/php56_e2e/web/grpc_pubsub.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Google\Auth\ApplicationDefaultCredentials;
+
+function updateAuthMetadataCallback($context)
+{
+    $auth_credentials = ApplicationDefaultCredentials::getCredentials(
+        ['https://www.googleapis.com/auth/pubsub']
+    );
+    return $auth_credentials->updateMetadata($metadata = [],
+                                             $context->service_url);
+}
+
+$channel_credentials = Grpc\ChannelCredentials::createSsl(
+    file_get_contents(__DIR__ . '/../vendor/grpc/grpc/etc/roots.pem'));
+
+$opts = ['credentials' => $channel_credentials,
+         'grpc.ssl_target_name_override' => 'pubsub.googleapis.com'];
+
+$client = new google\pubsub\v1\PublisherClient('pubsub.googleapis.com', $opts);
+
+$deadline = Grpc\Timeval::InfFuture();
+
+$req = new google\pubsub\v1\ListTopicsRequest();
+$proj = getenv('GAE_LONG_APP_ID');
+$req->setProject('projects/' . $proj);
+$call = $client->ListTopics(
+    $req,
+    [],
+    ['call_credentials_callback' => 'updateAuthMetadataCallback']);
```

Comment:
I believe the $service_url is used only in some xxxCredentials class, like `ServiceAccountJwtAccessCredentials` but not the others.

I like the inline function change.